src/components/BlocklyComponentWidget.svelte

```diff
@@ -0,0 +1,648 @@
+		const block = workspace.getBlockById($selected)!;
+		const component = getSelected()!;
+		if (component.summaryItems && component.summaryItems.length) {
+			component.summaryItems.forEach((item) => {
+				block.getField(item)?.setValue(`${item}: ${component.getValue(item)}`);
+			});
+		}
+		workspaceToSkill(workspace, skill);
+		if (onupdate) onupdate();
+	}
+</script>
+
+{#key [skill, $showSummaryItems, $useSymbols]}
+	<div class="wrapper">
+		<div style="height: 100%; flex-grow: 1;" use:blocklyInit></div>
+		{#if $selected}
+			{@const data = getSelected()!}
+			<div
+				class="selected-block"
+				style="height: 100%; width: 30em;"
+				transition:slide|local={{ axis: 'x', duration: 200 }}
+				onintrostart={function (e) {
+					const observer = new ResizeObserver(() => {
+						Blockly.svgResize(workspace);
+					});
+					observer.observe(e.target as Element);
+				}}
+			>
+				<div class="component-editor" style="width: 30em;">
+					<h2 class:deprecated={data.isDeprecated}><span>{data.name}</span></h2>
+					{#if data.description}
+						<div class="modal-desc">{@html data.description}</div>
+					{/if}
+					<hr />
+					<div class="component-entry">
+						<ProInput
+							label="Comment"
+							tooltip="[comment] A comment that will be displayed in the skill editor"
+							bind:value={data.comment}
+						/>
+						{#if data instanceof FabledTrigger && data.name !== 'Cast' && data.name !== 'Initialize' && data.name !== 'Cleanup'}
+							<BooleanSelectOption
+								name="Mana"
+								tooltip="[mana] Whether this trigger requires the mana cost to activate"
+								bind:data={data.mana}
+							/>
+							<BooleanSelectOption
+								name="Cooldown"
+								tooltip="[cooldown] Whether this trigger requires to be off cooldown to activate"
+								bind:data={data.cooldown}
+							/>
+						{:else if data instanceof FabledTarget || data instanceof FabledCondition || data instanceof FabledMechanic}
+							<ProInput
+								label="Icon Key"
+								bind:value={data.iconKey}
+								tooltip={'[icon-key] The key used by the component in the Icon Lore. If this is set to "example" and has a value name of "value", it can be referenced using the string "{attr:example.value}"'}
+							/>
+						{/if}
+						{#if data instanceof FabledMechanic}
+							<BooleanSelectOption
+								name="Counts as Cast"
+								tooltip={'[counts] Whether this mechanic running treats the skill as "casted" and will consume mana and start the cooldown. Set to false if it is a mechanic applled when the skill fails such as cleanup or an error message'}
+								bind:data={data.countsAsCast}
+							/>
+						{/if}
+
+						{#each data.data as datum}
+							{#if datum.meetsRequirements(data)}
+								<datum.component
+									bind:data={datum.data}
+									name={datum.name}
+									tooltip="{datum.key ? '[' + datum.key + '] ' : ''}{datum.tooltip}"
+									multiple={datum.multiple}
+								/>
+							{/if}
+						{/each}
```

Comment:
Options created in this section do not call $effect.pre(), resulting in dropdown options that require init not working as expected.

Reply:
Fixed in [29f7f38](https://github.com/magemonkeystudio/fabled/pull/1019/commits/29f7f380eebd5c3221e32a98f0cf3ff10d2185b1)